### PR TITLE
Support for vfl files and m3u improvements

### PR DESCRIPTION
**Pull request by monkeyman79**

### CODE REVIEW

File: libretro/libretro-core.c [hunk 1]
```diff
@@ -2149,17 +2157,20 @@ static bool retro_replace_image_index(unsigned index,
...
 		{
 			free(dc->files[index]);
 			dc->files[index] = NULL;
 		}
 		
-		// TODO : Handling removing of a disk image when info = NULL
-		
-		if(info != NULL) {
+		if (info != NULL)
+        {
```
Comment (rsn8887) [inline libretro/libretro-core.c]: Please fix indentation here and below

File: libretro/libretro-core.c [hunk 2]
```diff
@@ -2149,17 +2157,20 @@ static bool retro_replace_image_index(unsigned index,
...
 			dc->files[index] = strdup(info->path);
 		}
+        else
```
Comment (rsn8887) [inline libretro/libretro-core.c]: Please fix indentation

File: libretro/libretro-core.c [hunk 3]
```diff
@@ -2614,10 +2625,12 @@ size_t retro_serialize_size(void)
             fseek(file, 0L, SEEK_END);
             size = ftell(file);
             fclose(file);
+            unlink(file);
```
Comment (rsn8887) [inline libretro/libretro-core.c]: Please rebase your PR so these erroneous unlinks don’t appear in history, thanks.

File: libretro/libretro-core.c [hunk 4]
```diff
@@ -376,6 +264,257 @@ void parse_cmdline(const char *argv)
...
+    // Core command line is now parsed to ARGUV, ARGUC.
+    // Build command file for VICE in XARGV, PARAMCOUNT.
+
+    // If first command line argument is "x64", it's and extended command line
+    // otherwise it's just image filename
+    if (strcmp(ARGUV[0], CORE_NAME) != 0)
+    {
+        Add_Option(CORE_NAME);
+
+        // Ignore parser arguments, read filename directly from argv
```
Comment (monkeyman79) [inline libretro/libretro-core.c]:
```suggestion
        // Ignore parsed arguments, read filename directly from argv
```

**Comment by monkeyman79** [comment]
Should be good now.

**Comment by monkeyman79** [comment]
@rsn8887 btw. any ideas about that wiiu build failing?

**Comment by rsn8887** [comment]
Looks good, thanks!

**Comment by rsn8887** [comment]
I had to revert the last two commits, because af53027 causes crash on startup, see https://github.com/libretro/vice-libretro/issues/134

**Comment by rsn8887** [comment]
Maybe you can fix it and make a PR with a fixed version of those commits? Thanks.

**Comment by monkeyman79** [comment]
I'll look into that. I have some fixes in my sandbox, maybe it covers that
case.

śr., 18 wrz 2019, 20:03 użytkownik rsn8887 <notifications@github.com>
napisał:

**Comment by rsn8887** [comment]
I also made a new commit, removing the exit(1). I saw you removed it later as part of the two reverted commits. A core should not exit() ever, because it will crash all of Retroarch in static builds like Switch, Vita etc.

**Comment by rsn8887** [comment]
I made a version of your "Remove images from list" commit without conflicts, and pushed it.

Notice in your original commit, you deleted the file before closing once, on line 2662.

**Pull request merged**